lib/core/src/Cardano/Wallet/DB/SqliteTypes.hs

```diff
@@ -0,0 +1,242 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.DB.SqliteTypes where
+
+import Prelude
+
+import Control.Monad
+    ( (>=>) )
+import Data.Aeson
+import Data.Aeson.Types
+    ( Parser )
+import Data.Bifunctor
+    ( bimap, first )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Quantity
+    ( Quantity (..), getPercentage, mkPercentage )
+import Data.Text
+    ( Text )
+import Data.Text.Class
+    ( FromText (..)
+    , TextDecodingError (..)
+    , ToText (..)
+    , fromTextMaybe
+    , getTextDecodingError
+    )
+import Data.Word
+    ( Word64, Word8 )
+import Database.Persist.Sqlite
+    ( PersistField (..), PersistFieldSql (..), PersistValue )
+import GHC.Generics
+    ( Generic )
+import Web.HttpApiData
+import Web.PathPieces
+
+import qualified Data.Text as T
+
+import Cardano.Wallet.Primitive.Types
+    ( Coin (..)
+    , Direction (..)
+    , Hash (..)
+    , SlotId (..)
+    , TxStatus (..)
+    , WalletId (..)
+    , WalletState (..)
+    , flatSlot
+    , fromFlatSlot
+    , isValidCoin
+    )
```

Comment:
Imports are a bit messy here. `Data.Aeson`, `Web.HttpApiData` and `Web.PathPieces` are imported implicitly and `Cardano.Wallet.Primitive.Types` is detached from the rest ?

Reply:
OMG, I have addressed this issue already two days ago, but it seems my commit was los. correcting